### PR TITLE
refactored GARCH and added Mv(Gaussian/StudentT)RandomWalk

### PR DESCRIPTION
**Pull request by ferrine**
Added useful (I hope) distributions from my experiments and refactored garch a bit so it can be used as following
```python
GARCH11.dist(...).get_volatility(x)
```

### CODE REVIEW

**Comment by twiecki** [comment]
Does this fix https://github.com/pymc-devs/pymc3/issues/1491?

**Comment by ferrine** [comment]
No, but I'll make it so

**Comment by twiecki** [comment]
Actually that's probably a different problem. But do let me know if you have an idea.

File: pymc3/distributions/timeseries.py [hunk 1]
```diff
@@ -118,12 +127,12 @@ def volatility_update(x, vol, w, a, b):
...
                                      self.beta_1])
-        return vol
+        return tt.concatenate(self.initial_vol, vol)
 
     def logp(self, x):
-        vol = self._get_volatility(x[:-1])
-        return (Normal.dist(0., sd=self.initial_vol).logp(x[0]) +
-                tt.sum(Normal.dist(0, sd=vol).logp(x[1:])))
+        vol = self.get_volatility(x)
+        return (Normal.dist(0., sd=vol[0]).logp(x[0]) +
```
Comment (twiecki) [inline pymc3/distributions/timeseries.py]: Can't this be simplified now to: `tt.sum(Normal.dist(0, sd=vol).logp(x))`?

Comment (ferrine) [reply pymc3/distributions/timeseries.py]: Sure

**Comment by ferrine** [comment]
Ready to merge, #1491 is not related

**Review by springcoil — APPROVED** [review]

**Pull request merged**